### PR TITLE
[dynamo 3.11] enable dynamo unittests in 3.11

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96511
* __->__ #98104

Enable most dynamo unittests for 3.11. There are a few tests that are skipped due to failures that will be addressed in upcoming PRs.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @soumith @anijain2305 @Xia-Weiwen @peterbell10

### CODE REVIEW

**Review by yanboliang — APPROVED** [review]

**Review by voznesenskym — APPROVED** [review]

File: tools/dynamo/verify_dynamo.py [hunk 1]
```diff
@@ -215,9 +215,8 @@ def main():
         f"ROCM version: {rocm_ver}\n"
     )
     for args in _SANITY_CHECK_ARGS:
-        # TODO remove check when 3.11 is supported
-        if sys.version_info >= (3, 11):
-            warnings.warn("Dynamo not yet supported in Python 3.11. Skipping check.")
+        if sys.version_info >= (3, 12):
```
Comment (albanD) [inline tools/dynamo/verify_dynamo.py:218]: @e should leave a warning for 3.11 until everything is working fine no?

Comment (williamwen42) [reply tools/dynamo/verify_dynamo.py:218]: I'll leave a warning in eval_frame.py

**Review by albanD — APPROVED** [review]
SGTM when CI is green!

**Review by jansel — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Review by jerryzh168 — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Comment by williamwen42** [comment]
Skipping inductor minifier tests on Py 3.11+ since a bunch of the tests involve raising exceptions that aren't handled, which cause segfaults instead of a nicely printed stack trace.

**Pull request closed without merging**